### PR TITLE
Deepseek r1

### PR DESCRIPTION
**Pull request by sywangyi**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by sywangyi** [comment]
need 1.21 synapse AI rls

**Comment by sywangyi** [comment]
@regisss  in the PR,  MLA is optimized, in the same time fp8 kv cache is enabled/ warmup is accelerated for other models besides deepseek_r1.
we could merge the PR in advance and upgrade the dockerfile once 1.21 is rls to enable deepseek_R1

File: backends/gaudi/server/text_generation_server/models/flash_causal_lm.py [hunk 1]
```diff
@@ -1438,15 +1443,17 @@ def __init__(
...
         if htorch.utils.internal.is_lazy():
             htorch.hpu.wrap_in_hpu_graph(model, disable_tensor_cache=True)
         environment.set_model_config(self.config)
         self.use_contiguous_pa = (
             os.environ.get("VLLM_CONTIGUOUS_PA", "true").lower() == "true"
         )
-        self.limit_hpu_graphs = (
-            os.environ.get("LIMIT_HPU_GRAPHS", "false").lower() == "true"
+        self.limit_hpu_graph = (
+            os.environ.get("LIMIT_HPU_GRAPH", "false").lower() == "true"
```
Comment (regisss) [inline backends/gaudi/server/text_generation_server/models/flash_causal_lm.py:1454]: I don't think we should rename `LIMIT_HPU_GRAPHS` as it may already be used by some users.

File: backends/v3/src/queue.rs [hunk 2]
```diff
@@ -8,14 +8,14 @@ use std::cmp::max;
...
 use text_generation_router::infer::InferStreamResponse;
+use text_generation_router::usage_stats::Env;
 use text_generation_router::validation::{
     Chunk, ChunksToString, ValidGenerateRequest, ValidGrammar, ValidParameters,
     ValidStoppingParameters,
 };
 use tokio::sync::{mpsc, oneshot};
 use tokio::time::Instant;
 use tracing::{info_span, instrument, Instrument, Span};
-
```
Comment (regisss) [inline backends/v3/src/queue.rs:18]: Let's keep the original format for rust code and keep that blank line

File: backends/gaudi/server/text_generation_server/models/flash_causal_lm.py (hunk 1 above)
Comment (sywangyi) [reply backends/gaudi/server/text_generation_server/models/flash_causal_lm.py:1454]: yes. I just want to keep align with https://github.com/huggingface/text-generation-inference/blob/main/backends/gaudi/server/text_generation_server/models/causal_lm.py#L754, which is "LIMIT_HPU_GRAPH"

**Comment by regisss** [comment]
Pinging @Narsil for final approval

**Review by Narsil — APPROVED** [review]
